tendrl/commons/flows/create_cluster/utils.py

```diff
@@ -178,3 +180,83 @@ def gluster_create_ssh_setup_jobs(parameters, skip_current_node=False):
         )
     return ssh_job_ids
 
+
+def accuire_node_lock(parameters, flow_name):
+    Event(
+        Message(
+            job_id=parameters['job_id'],
+            flow_id=parameters['flow_id'],
+            priority="info",
+            publisher=NS.publisher_id,
+            payload={
+                "message": "Checking if nodes are not locked already"
+            }
+        )
+    )
+    fail = False
+    for node in parameters['Node[]']:
+        key = "/nodes/%s/locked_by" % node
+        try:
+            job_info = NS._int.client.read(key).value
+            fail = True
+            job_info = job_info.encode('utf8').replace("'", "\"")
+            job_info = json.loads(job_info)
+            Event(
+                Message(
+                    job_id=parameters['job_id'],
+                    flow_id=parameters['flow_id'],
+                    priority="error",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": "Node %s is already locked by a job %s" % \
+                            (
+                                node,
+                                job_info['job_id']
+                            )
+                    }
+                )
+            )
+        except EtcdKeyNotFound:
+            lock_info = dict(
+                job_id = str(parameters["job_id"]),
+                flow = flow_name
+            )
+            NS._int.client.write(key, lock_info)
+
+    if fail:
+        # Fail cluster creation flow
+        raise FlowExecutionFailedError(
+            "Few nodes are already locked by some job"
+        )
+    else:
+        Event(
+            Message(
+                job_id=parameters['job_id'],
+                flow_id=parameters['flow_id'],
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "All nodes are locked for cluster creation successfully"
+                }
+            )
+        )
+
+
+def release_node_lock(parameters):
+    Event(
+        Message(
+            job_id=parameters['job_id'],
+            flow_id=parameters['flow_id'],
+            priority="info",
+            publisher=NS.publisher_id,
+            payload={
+                "message": "Releasing node locks"
+            }
+        )
+    )
+    for node in parameters['Node[]']:
+        key = "/nodes/%s/locked_by" % node
+        try:
+            NS._int.client.delete(key)
+        except EtcdKeyNotFound:
```

Comment:
Why would there be a "EtcdKeyNotFound"?  if you are locking the nodes correctly

Remove this try/catch